packages/aztec.js/src/proof/publicRange/index.js

```diff
@@ -0,0 +1,174 @@
+const bn128 = require('@aztec/bn128');
+const { constants, errors, proofs } = require('@aztec/dev-utils');
+const BN = require('bn.js');
+const { AbiCoder } = require('web3-eth-abi');
+const { keccak256, padLeft } = require('web3-utils');
+
+const { inputCoder, outputCoder } = require('../../encoder');
+const helpers = require('./helpers');
+const { Proof, ProofType } = require('../proof');
+const ProofUtils = require('../utils');
+
+const { AztecError } = errors;
+
+class PublicRangeProof extends Proof {
+    /**
+     * Constructs a public range proof that a note is greater than or equal to, or less than or
+     * equal to a public integer. Control of whether a > or < proof is constructed is determined
+     * by an input boolean 'isGreaterOrEqual'.
+     *
+     * The balancing relation being satisfied is:
+     * originalNoteValue = publicComparison + utilityNoteValue
+     *
+     * @param {Object} originalNote the note that a user is comparing against the publicComparison
+     * @param {Number} publicComparison publicly visible integer, which the note is being compared against
+     * @param {string} sender Ethereum address of the transaction sender
+     * @param {bool} isGreaterOrEqual modifier controlling whether this is a greater than, or less
+     * than proof. If true, it is a proof that originalNoteValue > publicComparison. If false, it is a
+     * proof that originalNoteValue < publicComparison
+     * @param {Note} utilityNote a helper note that is needed to satisfy a cryptographic balancing relation
+     */
+    constructor(originalNote, publicComparison, sender, isGreaterOrEqual, utilityNote) {
+        const publicValue = constants.ZERO_BN;
```

Comment:
We should probably check to see if the inputs satisfy the proof relation, and throw an error if this is not the case

Reply:
Good idea, done